### ISSUE TITLE
Incorrect event in onChange callback when clearing an input using "clear icon"

### ISSUE THREAD

**Issue by Quernest** [lumada-design/hv-uikit-react#3424]
Hey guys!

Faced with the problem of clearing up an input using the icon. In this case `event` in the `onChange` is not correct because its targeting refers to the icon, not the event change in input.

<img width="749" alt="Screenshot 2023-06-13 at 14 03 56" src="https://github.com/lumada-design/hv-uikit-react/assets/21086792/fdf8fbb0-8bbf-42d2-b4cf-26f5d5265ce4">

The problem is that when processing `event.target.value` there is no valid value

What about providing additional prop `onClear` for custom clear implementations or (what is really better) fixing it via manually creating inputs `change` event with empty value

Haven't tested it, just an idea:
```ts
const event = new Event('input', { bubbles: true});
event.simulated = true;
inputRef.current.value = '';
inputRef.current.dispatchEvent(event);
```

or just modifying event
```ts
event.target.value = ''
onChange(event, '')
```

or smth. else :)

https://github.com/lumada-design/hv-uikit-react/blob/4ce3347087e3758fff78804b6a1448f19d747175/packages/core/src/components/Input/Input.tsx#LL556C1-L556C1

**Comment by zettca** [comment]
Hello @Quernest 👋 - thank you for reporting this 👍 

We definitely have an issue with the TS declarations, as we're not always returning a `ChangeEvent`.

Out of curiosity, why aren't you using the `onChange`'s returned `value` parameter? Ie:

```tsx
<HvInput onChange={(event, newValue) => console.log(newValue)} />
```

**Comment by Quernest** [comment]
> Hello @Quernest 👋 - thank you for reporting this 👍
> 
> We definitely have an issue with the TS declarations, as we're not always returning a `ChangeEvent`.
> 
> Out of curiosity, why aren't you using the `onChange`'s returned `value` parameter? Ie:
> 
> ```tsx
> <HvInput onChange={(event, newValue) => console.log(newValue)} />
> ```

Yes, this could certainly be a solution, but I it would cause custom implementations to handle `onChange`.

For example:
We have very popular form handling library [react-hook-form](https://www.react-hook-form.com/)

```tsx
import { useForm, Controller } from 'react-hook-form'
import { HvInput } from '@hitachivantara/uikit-react-core'

// instead of this
<Controller
  name="name"
  control={control}
  rules={{ required: true }}
  render={({ field }) => <HvInput {...field} />}
/>

// I have to do this
<Controller
  name="name"
  control={control}
  rules={{ required: true }}
  render={({ field: { onChange, ...otherFieldProps } }) => (
    <HvInput
      // create not a real event
      onChange={(event, value) => onChange({ target: { value } })}
      {...otherFieldProps}
    />
  )}
/>
```

Just because it handles `ChangeEvent` under the hood ... and it will work fine but to me it is not very convenient and looks like a crutch :)

**Comment by Quernest** [comment]
> Interesting - thanks for sharing the example! 👍
> 
> It definitely would make the usage simpler. We'll look into that.
> 
> There is also another issue with the use-case. `HvInput` isn't forwarding the `field.ref` - we probably should be. I'm unsure of the actual impact of this, but this warning will be raised:
> 
> ```
> Warning: Function components cannot be given refs. Attempts to access this ref will fail. Did you mean to use React.forwardRef()?
> ```
> 
> BTW: IIRC you can simply pass just the `value` to `onChange` - `react-hooks-form` handles that for you. ie:
> 
> ```tsx
> onChange={(event, value) => onChange(value)}
> ```
> 
> ([sample](https://stackblitz.com/edit/github-eyodit?file=src%2FApp.tsx))

Didn't know about it, thanks, that's a great point 👍
